[WIP] Support for on premise deployment

akram opened the pull request.

Commit:
Improving AWS support
 - Enabling the PersistControl mode which reduces execution time from 4min to 12s for an update
 - Changing calls to hostname command by the core hostname module
 - Changing calls to service command by the core service module
 - Adding accelerate: false for devug purposes
- Update README.md

jwhonce committed:
Infrastructure - Update pod definitions

* Bring pod definitions in line with lates k8s release

jwhonce committed:
Merge pull request #45 from jwhonce/wip/pod_updates

Infrastructure - Update pod definitions

L3n41c committed:
Explicitely use python2

Some distributions are using python3 as the default python.
On those ones, we need to explicitely use python2.

L3n41c committed:
Make the AWS region, AMI, keypair, instance type and OpenShift binary customisable

The AWS region, AMI, keypair and instance type are not hardcoded anymore.
They can be customized in the config file.
If the config file is empty, the default values are the previous hardcoded ones.
It is also possible to specify a local OpenShift binary to deploy.
If not specified, it will intall one from the yum repository.

L3n41c committed:
Remove useless docker install on openshift master

L3n41c committed:
Add an "overlay network" for the inter-POD communication

It is built thanks to Open vSwitch and VxLAN.

The network topology is statically built and is described in
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/ovs-networking.md
except that each node gets a 192.168.x.0/24 subnet where x is the node index.

L3n41c committed:
Split the VM launch from the VM configuration

As `./cluster.sh` already launches `config.yml` playbook after having launched the `launch.yml` one,
it is useless to include `config.yml` at the end of `launch.yml`.

The expected impact of this patch is cluster creation time improvement.

Commit:
The minion index is now held by a local (remote) fact

L3n41c committed:
Skip installation of rhel 7 libra candidate repo if we deploy a locally built OpenShift

If we want to deploy a locally built OpenShift, we don’t need this repo.
The goal is the make these scripts as independant from libra as possible.

L3n41c committed:
Cosmetic yaml fix

L3n41c committed:
Fix VxLAN creation

When we create the VxLAN links between minions, we have to loop over all the minions
and to skip ourself. A minion doesn’t need a link to itself.

The "groups['tag_env-host-type_' + oo_env + '-openshift-minion']" over which we iterate contains the public IPs.
So, we have to use facter_ec2_public_ipv4 to get the public IP of the current minion when doing the filtering.

L3n41c committed:
Use ~/.awscred instead of ~/aws-credentials

L3n41c committed:
Remove "accelerate: false"

Accelerate mode is not enabled by default and is deprecated by SSH Pipelining.

L3n41c committed:
Factorize the OpenShift configuration

L3n41c committed:
Create a config.sample file

This file can be used as a base to create a personal config file.
config.sample is committed and is not meant to be modified by users.
config is created and modified by users, but is not committed in git.

L3n41c committed:
Fix a bug in the list command when some VMs have no tag name

L3n41c committed:
Specify that we tested those scripts on RHEL7

L3n41c committed:
Fixes for PR #46

L3n41c committed:
Re-include config.yml from launch.yml but make it optional

It is not possible to first launch and configure the master and then to launch
and configure the minions because the master needs to know the IPs of the
minions (in the --nodes parameter found in /etc/sysconfig/openshift)

If we proceed in this order:
* launch the master machine
* configure the master
* launch the minion machines
* configure the minions
what happens is that the "--nodes" parameter of openshift is empty.
This prevents openshift master from starting.
Then the minions are starting, but they fail because they were enable to
connect to the master.
At the end, neither openmaster nor openshift minions are running.

In order to have things properly configured, i.e. to have the master know
the minions IPs and the minions knows the master IP, we need to proceed in
this order:
* launch the master machine
* launch the minion machines
* configure the master
* configure the minion

The "config" tag is used to skip the configuration step when launching
the machines. cluster.sh already takes care of doing the "config" step
after the "launch" one.

Commit:
Allows it to run on OSX
- Versions of ruby ported on OSX requires escaping of hyphens in regexp
- Adds documentation on packages required by OSX to run openshift-online-ansible on OSX

L3n41c committed:
Assign an index to minions for container subnet

Each machine hosting a minion has two local ansible facts:

* subnet_index which must be different between each machine. It is assigned at
               creation (setup from launch.yml)

* nb_minions   keeps the current number of minions in the cluster
               It is used when spawning new minions on an existing cluster
               to guarantee that the new machines will get unused indices.

L3n41c committed:
Merge remote-tracking branch 'akram/osx' into aws_openvswitch_merged_with_akram

L3n41c committed:
Remove the double installation of .vimrc

L3n41c committed:
Remove the ad-hoc Ansible filter and use oo_collect instead

Commit:
Support local deployment through local inventory
- Support local deployment on two ansible groups named 'masters' and 'minions'
- Adds support for certificate authentication

Commit:
Support local deployment through local inventory
- Support local deployment on two ansible groups named 'masters' and 'minions'
- Adds support for certificate authentication

L3n41c committed:
Misc fixes

lhuard1A commented:
@akram :
I have rebased your akram/on_premise branch on top of lhuard1A/aws_openvswitch (which is the [PR #46](https://github.com/openshift/openshift-online-ansible/pull/46). The result is the branch lhuard1A/on_premise.

akram commented:
Thanks 
I will have a look at this

Sent from mobile

> On 13 févr. 2015, at 16:50, Lénaïc Huard notifications@github.com wrote:
> 
> @akram :
> I have rebased your akram/on_premise branch on top of lhuard1A/aws_openvswitch (which is the PR #46. The result is the branch lhuard1A/on_premise.
> 
> —
> Reply to this email directly or view it on GitHub.

Commit:
Merge remote-tracking branch 'lhuard/on_premise' into on_premise

Conflicts:
	playbooks/lan/openshift-minion/launch.yml

Commit:
On premise deployment
- Reverting to tag_env_host_type* group name for coherence with other playbooks
- Removing local_openshift_* roles to reuse the openshift_master and openshift_minion roles
- Adding docker systemd unit to create socket on non-RHEL distribution of docker
- Adds CentOS repository to get docket
- Update documentation

L3n41c committed:
Fix miscellaneous issues

Commit:
Merge pull request #4 from lhuard1A/on_premise

Fix miscellaneous issues

Commit:
- Adding helloworld openshift example
- Fixing documentation
- Adding template for openshift-router
- Lowering loglevel
- Adding restart after openshift binary copy
